src/Trial.hs

```diff
@@ -377,3 +389,127 @@ instance
             Fiasco e          -> Fiasco e
             Result e (tag, a) -> Result (DL.snoc e $ fieldName <> tag) a
     {-# INLINE fromLabel #-}
+
+{- $patternList
+
+>>> :{
+foo :: Trial String Int -> String
+foo (FiascoL []) = "Fiasco list is empty"
+foo (ResultL [] _) = "Result list is empty"
+foo _ = "Other case"
+:}
+
+>>> foo $ fiascos []
+"Fiasco list is empty"
+>>> foo $ pure 42
+"Result list is empty"
+>>> foo $ result "Something" 42
+"Other case"
+-}
+
+{- |
+
+@since 0.0.0.0
+-}
+pattern FiascoL :: [(Fatality, e)] -> Trial e a
+pattern FiascoL e <- Fiasco (DL.toList -> e)
+
+{- |
+
+@since 0.0.0.0
+-}
+pattern ResultL :: [e] -> a -> Trial e a
+pattern ResultL e a <- Result (DL.toList -> e) a
```

Comment:
I just thought that our `Warning` and `Error` patterns are complete. So we can add complete pragma for them, like so:

```haskell
{-# COMPLETE Warning, Error #-}
```

I guess we can do the same for `FiascoL` and `ResultL`